src/azul/project/hca/transformers.py

```diff
@@ -490,16 +518,15 @@ def field_types(cls) -> FieldTypes:
         }
 
 
-def _parse_zarr_file_name(file_name):
-    delimiter = [delim for delim in ('.zarr!', '.zarr/') if delim in file_name]
```

Comment:
@amarjandu your version failed tests if I removed the `.zarr!` case. That indicated that the ! substitution was incomplete in your version, applied too late in the control flow and missed some branches. The introduction of the `_transform` method consolidates the ! substitution into one place and is a net win wrt SLOCs (-11).